src/work/PokerGame.sol

```diff
@@ -0,0 +1,226 @@
+        require(game.state == GameState.WAITING, "Game not in waiting state"); // Ensure the game is in the WAITING state
+        require(game.playerCount < MAX_PLAYERS, "Game is full"); // Ensure the game is not full
+
+        // Add the player to the game
+        game.players[game.playerCount] = Player({
+            addr: msg.sender,
+            chipCount: INITIAL_CHIP_COUNT,
+            lastAction: PlayerAction.NONE,
+            isActive: true
+        });
+        game.playerCount++; // Increment the player count
+
+        emit PlayerJoined(gameId, msg.sender); // Emit the PlayerJoined event
+
+        // If the game is full, start the game
+        if (game.playerCount == MAX_PLAYERS) {
+            startGame(gameId);
+        }
+    }
+
+    // Function to start the game
+    function startGame(uint256 gameId) public {
+        Game storage game = games[gameId]; // Get the game from storage
+        require(game.state == GameState.WAITING, "Game not in waiting state"); // Ensure the game is in the WAITING state
+        require(game.playerCount >= 2, "Not enough players"); // Ensure there are at least 2 players
+
+        game.state = GameState.ACTIVE; // Set the game state to ACTIVE
+        game.dealerIndex = 0; // Set the dealer index to 0
+        game.smallBlindIndex = (game.dealerIndex + 1) % game.playerCount; // Set the small blind index
+        game.bigBlindIndex = (game.smallBlindIndex + 1) % game.playerCount; // Set the big blind index
+        game.currentPlayerIndex = (game.bigBlindIndex + 1) % game.playerCount; // Set the current player index
+
+        // Collect blinds from the small and big blind players
+        collectBlind(gameId, game.smallBlindIndex, 1);
+        collectBlind(gameId, game.bigBlindIndex, 2);
+
+        emit GameStarted(gameId); // Emit the GameStarted event
+        emit TurnChanged(gameId, game.players[game.currentPlayerIndex].addr); // Emit the TurnChanged event
+    }
+
+    // Internal function to collect blinds from a player
+    function collectBlind(uint256 gameId, uint8 playerIndex, uint256 blindAmount) internal {
+        Game storage game = games[gameId]; // Get the game from storage
+        Player storage player = game.players[playerIndex]; // Get the player from the game
+
+        require(player.chipCount >= blindAmount, "Player doesn't have enough chips for blind"); // Ensure the player has enough chips
+
+        player.chipCount -= blindAmount; // Deduct the blind amount from the player's chip count
+        game.pot += blindAmount; // Add the blind amount to the pot
+        game.highestBet = (blindAmount > game.highestBet) ? blindAmount : game.highestBet; // Update the highest bet if necessary
+
+        emit PlayerAction(gameId, player.addr, PlayerAction.CALL, blindAmount); // Emit the PlayerAction event
+    }
+
+    // Function for a player to place a bet
+    function placeBet(uint256 gameId, uint256 amount) external nonReentrant {
```

Comment:
we can create a role that gets granted to the address when its their turn and so that way we dont have to do the msg.sender check